File: artifacts/nexus-plus/android/app/src/main/java/com/nexuswavetech/nexusplus/VocalRemoverNative.java
package com.nexuswavetech.nexusplus;

/**
 * Minimal JNI surface for the native C++ vocal-removal engine.
 * JS/TS should call this layer only from a development/native Android build.
 */
public final class VocalRemoverNative {
  static {
    System.loadLibrary("nexus_vocal_remover");
  }

  private VocalRemoverNative() {}

  public static native boolean isAvailable();

  /** Returns output path on success or an error message on failure. */
  public static native String separate(
      String inputPath,
      String outputPath,
      String stem,
      String quality,
      boolean preserveBass,
      boolean preserveStereo,
      int chunkSeconds
  );

  public static native void cancel();
  public static native void dispose();
}

File: artifacts/nexus-plus/native/vocal_remover/VocalRemoverNative.kt
package com.nexuswavetech.nexusplus.vocal

object VocalRemoverNative {
    init {
        System.loadLibrary("nexus_vocal_remover")
    }

    @JvmStatic
    external fun nativeIsAvailable(): Boolean

    @JvmStatic
    external fun nativeSeparate(
        inputPath: String,
        outputPath: String,
        quality: Int,
        preserveBass: Boolean,
        preserveStereo: Boolean,
    ): String?

    @JvmStatic
    external fun nativeCancel()

    @JvmStatic
    external fun nativeDispose()
}
